jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -1590,18 +1590,18 @@ public void failed(Throwable failure)
 
                     httpChannelState._callbackFailure = failure;
 
-                    // Consume any input.
-                    Throwable unconsumed = stream.consumeAvailable();
-                    ExceptionUtil.addSuppressedIfNotAssociated(failure, unconsumed);
+                    if (!stream.isCommitted() && !(failure instanceof Request.Handler.AbortException))
```

Comment:
Before, `stream.consumeAvailable()` was called in any case, now only if the stream is not committed.
I feel like we should restore the old behavior.

Reply:
@sbordet Why do we want to waste time/effort consuming content when we are aborting/closing/resetting?

Reply:
Good point.